wazo_call_logd/sync_db.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import argparse
+import logging
+
+from xivo import xivo_logging
+from xivo.chain_map import ChainMap
+from xivo.config_helper import read_config_file_hierarchy
+from xivo_dao import init_db_from_config
+from wazo_auth_client import Client as AuthClient
+from wazo_call_logd.database.models import Tenant
+from wazo_call_logd.database.helpers import new_db_session
+from wazo_call_logd.database.queries.tenant import TenantDAO
+from wazo_call_logd.config import DEFAULT_CONFIG, _load_key_file
+from wazo_call_logd.purger import CallLogsPurger, ExportsPurger, RecordingsPurger
+
+logger = logging.getLogger('wazo-call-logd-sync-db')
+
+
+def parse_cli_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        help="Log debug messages",
+    )
+    parser.add_argument(
+        '-q',
+        '--quiet',
+        action='store_true',
+        help='Only print warnings and errors',
+    )
+    parsed_args = parser.parse_args()
+    result = {'log_level': logging.INFO}
+    if parsed_args.quiet:
+        result['log_level'] = logging.WARNING
+    elif parsed_args.debug:
+        result['log_level'] = logging.DEBUG
+    return result
+
+
+def load_config():
+    file_config = read_config_file_hierarchy(ChainMap(DEFAULT_CONFIG))
+    service_key = _load_key_file(ChainMap(file_config, DEFAULT_CONFIG))
+    return ChainMap(service_key, file_config, DEFAULT_CONFIG)
+
+
+def main():
+    from time import sleep
+
+    sleep(20)
+    cli_args = parse_cli_args()
+    config = load_config()
+
+    xivo_logging.setup_logging('/dev/null', log_level=cli_args['log_level'])
+    xivo_logging.silence_loggers(['stevedore.extension'], logging.WARNING)
+
+    token = AuthClient(**config['auth']).token.new(expiration=300)['token']
+
+    del config['auth']['username']
+    del config['auth']['password']
+    tenants = AuthClient(token=token, **config['auth']).tenants.list()['items']
```

Comment:
Not sure what's happening here.

Reply:
What do you mean ?
The same thing is done in the syncdb of wazo-confd :  https://github.com/wazo-platform/wazo-confd/blob/8204008a1f15b44f6ae04dda5a8ececa27d39a58/wazo_confd/sync_db.py#L66

Reply:
Why are the authentication credentials dismissed for this call of AuthClient, but used above?

Is this a necessary useful pattern, or just a historical artifact?

Either way, I think this deserves a comment because it's not obvious from the code.

Reply:
The auth client is only created only with a token, this is why any potential username/password are deleted before.

Reply:
Ah right, the auth credentials is used to get a token, which is then used for subsequent auth requests. Sorry didn't have the auth flow in my head.